common/metrics/client.go

```diff
@@ -29,10 +30,11 @@ import (
 // ClientImpl is used for reporting metrics by various Cadence services
 type ClientImpl struct {
 	//parentReporter is the parent scope for the metrics
-	parentScope tally.Scope
-	childScopes map[int]tally.Scope
-	metricDefs  map[int]metricDefinition
-	serviceIdx  ServiceIdx
+	parentScope  tally.Scope
+	childScopes  map[int]tally.Scope
+	metricDefs   map[int]metricDefinition
+	serviceIdx   ServiceIdx
+	cachedScopes map[string]Scope
```

Comment:
how is this thread safe ?

Reply:
Its not right now, I need to think through this thread safety still.